Lib/tarfile.py

```diff
@@ -2697,13 +2694,10 @@ def next(self):
             except SubsequentHeaderError as e:
                 raise ReadError(str(e)) from None
             except Exception as e:
-                try:
-                    import zlib
-                    if isinstance(e, zlib.error):
-                        raise ReadError(f'zlib error: {e}') from None
-                    else:
-                        raise e
-                except ImportError:
+                import zlib
+                if isinstance(e, zlib.error):
+                    raise ReadError(f'zlib error: {e}') from None
+                else:
```

Comment:
For *this* one, it might be better to keep the try/except. IMO, reading *uncompressed* tarfiles should work as before, even on WASI.